### PR TITLE
Add the introduction of capture group for regexp function

### PR DESCRIPTION
**Pull request by xzhangxian1008**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/string-functions.md [hunk 1]
```diff
@@ -1485,9 +1485,9 @@ TiDB 与 MySQL 在二进制字符串 (binary string) 数据类型上的差异：
...
-TiDB 与 MySQL 在替换空字符串上的差异：
+- TiDB 与 MySQL 在替换空字符串上存在差异，下面以 `REGEXP_REPLACE("", "^$", "123")` 为例：
 
-下面以 `REGEXP_REPLACE("", "^$", "123")` 为例：
+  - MySQL 不会对空串进行替换，其结果为 `""`。
+  - TiDB 会对空串进行替换，其结果为 `"123"`。
 
-- MySQL 不会对空串进行替换，其结果为 `""`。
-- TiDB 会对空串进行替换，其结果为 `"123"`。
+- TiDB 与 MySQL 在捕获组的关键字上存在差异。Mysql 的捕获组关键字为`$`，而 TiDB 的捕获组关键字为`\\`，同时 TiDB 只支持编号为 0-9 的捕获组。
```
Comment (windtalker) [inline functions-and-operators/string-functions.md:1493]: 这个需要加一个具体例子吗？

Comment (xzhangxian1008) [reply functions-and-operators/string-functions.md]: > 这个需要加一个具体例子吗？

加好了

**Review by windtalker — APPROVED** [review]
LGTM

File: functions-and-operators/string-functions.md [hunk 2]
```diff
@@ -1485,9 +1485,11 @@ TiDB 与 MySQL 在二进制字符串 (binary string) 数据类型上的差异：
...
 
-下面以 `REGEXP_REPLACE("", "^$", "123")` 为例：
+  - MySQL 不会对空串进行替换，其结果为 `""`。
+  - TiDB 会对空串进行替换，其结果为 `"123"`。
 
-- MySQL 不会对空串进行替换，其结果为 `""`。
-- TiDB 会对空串进行替换，其结果为 `"123"`。
+- TiDB 与 MySQL 在捕获组的关键字上存在差异。Mysql 的捕获组关键字为`$`，而 TiDB 的捕获组关键字为`\\`，同时 TiDB 只支持编号为 0-9 的捕获组。例如：
+
+  - `select regexp_replace('abcd','(.*)(.{2})$','\\1') as s;`返回结果`ab`
```
Comment (qiancai) [inline functions-and-operators/string-functions.md:1495]:
```suggestion
- TiDB 与 MySQL 在捕获组的关键字上存在差异。MySQL 的捕获组关键字为`$`，而 TiDB 的捕获组关键字为`\\`。此外，TiDB 只支持编号为 0 到 9 的捕获组。

    例如，以下 SQL 语句在 TiDB 中的返回结果为 `ab`。

    ```sql
    SELECT REGEXP_REPLACE('abcd','(.*)(.{2})$','\\1') AS s;
    ```

**Review by qiancai — APPROVED** [review]

**Review by qiancai — APPROVED** [review]

**Pull request merged**